jni/tests/faiss_memory_test.cpp

```diff
@@ -0,0 +1,298 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+#include "faiss_wrapper.h"
+#include "nmslib_wrapper.h"
+
+#include <vector>
+#include <malloc.h>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "jni_util.h"
+#include "test_util.h"
+#include "faiss/utils/utils.h"
+
+using ::testing::NiceMock;
+using ::testing::Return;
+#define GTEST_COUT std::cerr << "[          ] [ INFO ]"
+
+TEST(FaissHNSWIndexMemoryTest, BasicAssertions) {
+
+    char dataset[] = "dataset/sift/sift_base.fvecs";
+    float* data_load = NULL;
+    unsigned points_num, dim;
+    test_util::load_data(dataset, data_load, points_num, dim);
+
+    GTEST_COUT << "points_num："<< points_num << " data dimension：" << dim << std::endl;
+    float* dataptr = data_load;
+
+    faiss::idx_t numIds = points_num;
+    std::vector<faiss::idx_t> ids(points_num);
+    std::vector<std::vector<float>> vectors;
+    for (int64_t i = 0; i < numIds; ++i) {
+        ids[i] = i;
+
+        std::vector<float> vect;
+        vect.reserve(dim);
+        for (int j = 0; j < dim; ++j) {
+            vect.push_back(*dataptr);
+	    dataptr++;
+        }
+        vectors.push_back(vect);
+    }
+
+    free(data_load);
+    malloc_trim(0);
+
+    std::string indexPath = "tmp/FaissHNSWIndexMemoryTest.faiss";
+    std::string spaceType = knn_jni::L2;
+    std::string index_description = "HNSW32,Flat";  
+    int thread_num = 7;
```

Comment:
why 7 threads?

Reply:
cause my dev environment is 8 core, using 7 thread to build index, and 1 core for monitoring